### PR TITLE
Document the new field 'basetime' for the timestamp generator

### LINKED ISSUE #693: Document the new field `basetime` for the load generator connector for generating timestamp data
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/8619

### Which part(s) of the docs might be affected or should be updated? And how?

Data ingestion -> Load generator
Note this field is only valid for timestamp data.

### Reference

_No response_

### PR DESCRIPTION
**Pull request by CharlieSYH**
…generating timestamp data

## Info

- **Preview**: 
https://pr-731.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/create-source-datagen/

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/pull/8619

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/693

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/create-source/create-source-datagen.md [hunk 1]
```diff
@@ -149,7 +149,8 @@ Specify the following fields for every column in the source you are creating.
...
 |`max_past`|Specify the maximum deviation from the baseline timestamp to determine the earliest possible timestamp can be generated. |An [interval](/sql/sql-data-types.md)<br/>Example: `2h 37min`|False<br/>Default: `1 day`|
-|`max_past_mode`|Specify the baseline timestamp. <br/> The range for generated timestamps is [baseline - `max_past` , baseline]|`absolute` — Baseline is set to the creation time of the source.<br />`relative` —  Baseline is the current system time.|False<br/>Default: `absolute`|
+|`max_past_mode`|Specify the baseline timestamp. <br/> The range for generated timestamps is [basetime - `max_past` , basetime]|`absolute` — Baseline is set to the execution time of the generator.<br />`relative` —  Baseline is the current system time.|False<br/>Default: `absolute`|
+|`basetime`|If set, the generator will use the specified time as the basetime. If omitted, the generator execution time is set as the basetime.|A [date and time string](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.parse_from_rfc3339)<br/>Example: `2023-04-01T16:39:57-08:00`|False<br/>Default: generator execution time|
```
Comment (hengm3467) [inline docs/create-source/create-source-datagen.md]: This is a minor thing. I suggest change "omitted" to "unspecified". The difference is subtle but "unspecified" is neutral.

**Review by hengm3467 — APPROVED** [review]
Otherwise looks good to me.

File: docs/create-source/create-source-datagen.md [hunk 2]
```diff
@@ -149,7 +149,8 @@ Specify the following fields for every column in the source you are creating.
 |---|---|---|---|
 |`kind`|Generator type|Set to `random`.|False<br/>Default: `random`|
 |`max_past`|Specify the maximum deviation from the baseline timestamp to determine the earliest possible timestamp can be generated. |An [interval](/sql/sql-data-types.md)<br/>Example: `2h 37min`|False<br/>Default: `1 day`|
-|`max_past_mode`|Specify the baseline timestamp. <br/> The range for generated timestamps is [baseline - `max_past` , baseline]|`absolute` — Baseline is set to the creation time of the source.<br />`relative` —  Baseline is the current system time.|False<br/>Default: `absolute`|
+|`max_past_mode`|Specify the baseline timestamp. <br/> The range for generated timestamps is [basetime - `max_past` , basetime]|`absolute` — Baseline is set to the execution time of the generator.<br />`relative` —  Baseline is the current system time.|False<br/>Default: `absolute`|
```
Comment (tabVersion) [inline docs/create-source/create-source-datagen.md]: `relative` —  Baseline is the current system time. => `relative` —  Baseline is the system time everytime generates a new record.

File: docs/create-source/create-source-datagen.md [hunk 3]
```diff
@@ -149,7 +149,8 @@ Specify the following fields for every column in the source you are creating.
...
+|`basetime`|If set, the generator will use the specified time as the basetime. If unspecified, the generator execution time is set as the basetime.|A [date and time string](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.parse_from_rfc3339)<br/>Example: `2023-04-01T16:39:57-08:00`|False<br/>Default: generator execution time|
```
Comment (tabVersion) [inline docs/create-source/create-source-datagen.md]: override `max_past_mode`, the fixed basetime can be shared across exec.

**Review by tabVersion — APPROVED** [review]
rest LGTM

**Pull request merged**